From 7dfc8d94434d759443d724bce76e8b47fb8986e8 Mon Sep 17 00:00:00 2001
From: Martin Pitt <mpitt@debian.org>
Date: Tue, 9 Aug 2016 18:11:25 +0200
Subject: Don't link libcups.so and libcupsimage.so against unnecessary
 libraries.

 This avoids unnecessary package dependencies for
 both libcups, as well as for packages using cups-config.
Bug: https://www.cups.org/str.php?L4343
Bug-Debian: http://bugs.debian.org/438067

Patch-Name: drop_unnecessary_dependencies.patch
---
 Makedefs.in   | 2 +-
 cups/Makefile | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makedefs.in b/Makedefs.in
index f96a5e9bf..d3e96e8e3 100644
--- a/Makedefs.in
+++ b/Makedefs.in
@@ -157,7 +157,7 @@ DBUS_NOTIFIERLIBS =	@DBUS_NOTIFIERLIBS@
 DNSSD_BACKEND	=	@DNSSD_BACKEND@
 DSOFLAGS	=	-L../cups @ARCHFLAGS@ @LDFLAGS@ @RELROFLAGS@ \
 			@DSOFLAGS@ $(OPTIM)
-DSOLIBS		=	@DSOLIBS@ $(COMMONLIBS)
+DSOLIBS		=	@DSOLIBS@
 DNSSDLIBS	=	@DNSSDLIBS@
 IPPFIND_BIN	=	@IPPFIND_BIN@
 IPPFIND_MAN	=	@IPPFIND_MAN@
diff --git a/cups/Makefile b/cups/Makefile
index b56e5f34a..9322b9d93 100644
--- a/cups/Makefile
+++ b/cups/Makefile
@@ -356,7 +356,7 @@ uninstall:
 libcups.so.2:	$(LIBOBJS)
 	echo Linking $@...
 	$(DSO) $(ARCHFLAGS) $(DSOFLAGS) -o $@ $(LIBOBJS) \
-		$(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) $(COMMONLIBS) $(LIBZ)
+		$(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) -lpthread $(LIBZ) -lm
 	$(RM) `basename $@ .2`
 	$(LN) $@ `basename $@ .2`
 
@@ -372,7 +372,7 @@ libcups.2.dylib:	$(LIBOBJS)
 		-current_version 2.14.0 \
 		-compatibility_version 2.0.0 \
 		$(LIBOBJS) \
-		$(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) $(COMMONLIBS) $(LIBZ)
+		$(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) -lpthread $(LIBZ)
 	$(CODE_SIGN) -s "$(CODE_SIGN_IDENTITY)" $@
 	$(RM) libcups.dylib
 	$(LN) $@ libcups.dylib
@@ -423,7 +423,7 @@ libcups2.def: $(LIBOBJS) $(IMAGEOBJS) Makefile
 
 libcupsimage.so.2:	$(IMAGEOBJS) libcups.so.2
 	echo Linking $@...
-	$(DSO) $(ARCHFLAGS) $(DSOFLAGS) -o $@ $(IMAGEOBJS) $(DSOLIBS) $(LINKCUPS)
+	$(DSO) $(ARCHFLAGS) $(DSOFLAGS) -o $@ $(IMAGEOBJS) $(DSOLIBS) $(LINKCUPS) -lpthread
 	$(RM) `basename $@ .2`
 	$(LN) $@ `basename $@ .2`
 
